drv/lpc55-update-server/src/main.rs

```diff
@@ -110,34 +112,27 @@ impl idl::InOrderUpdateImpl for ServerImpl {
             return Err(UpdateError::BadLength.into());
         }
 
-        let mut flash_page: [u8; BLOCK_SIZE_BYTES] = [0; BLOCK_SIZE_BYTES];
+        let mut flash_page = [0u8; BLOCK_SIZE_BYTES];
 
-        block
-            .read_range(0..len as usize, &mut flash_page)
-            .map_err(|_| RequestError::Fail(ClientError::WentAway))?;
+        if block_num == HEADER_BLOCK {
+            self.block0 = Some([0u8; BLOCK_SIZE_BYTES]);
+            let block0 = &mut self.block0.as_mut().unwrap_lite()[..];
```

Comment:
This seems odd to assign the block and then immediately unwrap it, is this necessary?

Reply:
I agree that it looks ridiculous. I think if I don't do this though, the compiler may perform a stack copy resulting in double size. I've seen a stack overflow due to something similar before.

Reply:
Ok, I will experiment and see if it makes a difference with stack margin if I do the copy before inserting into self.

Reply:
So a separate boolean may be better than Option<>?

Reply:
I think we can actually just do it with a map without the unwrap. Using a bool makes me feel icky.

Reply:
Update on this, copying the array into self did cause a stack overflow. I'm still playing around.

Reply:
Ah, there is an entry style API, which is exactly what we want. https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert

Reply:
fixed